### ISSUE TITLE
Django Ticket 16752: Multi-db without a 'default' database

### ISSUE THREAD

**Issue by ryancheley** [ryancheley/public-notes#5]
I decided to pick up a ticket [16752](https://code.djangoproject.com/ticket/16752) in Django Trac 

The content of the ticket is:

In the [https://docs.djangoproject.com/en/dev/topics/db/multi-db multi-db doc] it says:
```
If you don’t have a default database, you need to be careful to always specify the database that you want to use.
```

I have a project where that is exactly the behavior I'd prefer - I want to explicitly route every call and have nothing go to the 'default' database.

However, when I removed the 'default' key from DATABASES (and hard-coded a router to always return a different value), model validation failed.  This seems to be because it imports
```
from django.db import connection
```
then proceeds to validation the schema against that connection (which is, in fact, the nonexistent 'default').  This fails as soon as an operation is called on the connection.

There is a second problem with this -- all model validation occurs against the default connection rather than the connection that the model will be routed to.  That seems like a bug, but I'd like to treat it separately.

My intention with this ticket is to establish: do we intend projects to work without a 'default' project, as implied by the docs, or is a 'default' database always assumed?

**Comment by ryancheley** [comment]
If I'm understanding the report correctly, the underlying issue is that IF you wanted to use anything OTHER than the default database in a multi-db environment, you can't because the underpinnings assume a database called `default`. I do have a bit of a work around to allow a user to specify the name of the default database to be something OTHER than default, but I'm not sure if that actually solve the problem, or just hides it.

**Comment by ryancheley** [comment]
So after a bit of a work I did a PR to get a bit more feedback, and I don’t really understand the underlying issue well enough to be able to tackle this. There is also another issue that is really needs to be completed FIRST, so this was a good attempt to jump in the water, but I may have both jumped into the deep end AND upon jumping, realized that the pool was frozen over. 

I’m going to attempt to take a look at the blocking ticket, but in my cursory reading of that, I’m not sure I’ll be able to move it forward either, but I’m going to give it a shot!

**Comment by ryancheley** [comment]
In reviewing ticket [13528](https://code.djangoproject.com/ticket/13528) the last [comment](https://code.djangoproject.com/ticket/13528#comment:22) makes me think that this really isn't something I can do either

> I discussed this issue a little with Loic and this is what he suggested that we can note down where and why it is possible to change the suggested behaviour and where all the alias is not used and should be so that we can move on from this and try to **fix that bigger problem** instead of this one and that might take considerable amount of time.

I'm still struggling to understand what the **bigger** problem is though.
